dashboard/app/helpers/levels_helper.rb

```diff
@@ -471,6 +475,44 @@ def firebase_options
     fb_options
   end
 
+  def azure_speech_service_options
+    speech_service_options = {}
+
+    if @level.game.use_azure_speech_service?
+      uri = URI.parse("https://#{CDO.azure_speech_service_region}.api.cognitive.microsoft.com/sts/v1.0/issueToken")
+      header = {'Ocp-Apim-Subscription-Key': CDO.azure_speech_service_key}
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+      request = Net::HTTP::Post.new(uri.request_uri, header)
+      response = http.request(request)
+      speech_service_options[:azureSpeechServiceToken] = response.body
+      speech_service_options[:azureSpeechServiceRegion] = CDO.azure_speech_service_region
+      uri = URI.parse("https://#{CDO.azure_speech_service_region}.tts.speech.microsoft.com/cognitiveservices/voices/list")
+      header = {'Authorization': 'Bearer ' + response.body}
+      http = Net::HTTP.new(uri.host, uri.port)
+      http.use_ssl = true
+      http.verify_mode = OpenSSL::SSL::VERIFY_PEER
```

Comment:
Looks like some of these are repeated from a few lines above. Is that intentional?

Reply:
its intentional. We need to make a request for an access token first and then use that token to request the list of voices and languages.

Reply:
Ahh makes sense. In that case, would you mind splitting them into two sections and using different names for the two sets of variables?
i.e.
```
// First, get the token
tokenUri = ...
tokenHeader = ...
tokenHttpRequest = ...

// Then, get the list of voices and languages
voiceUri = ...
voiceHeader = ...
 ...
```